time.c

```diff
@@ -3331,36 +3331,43 @@ tmcmp(struct tm *a, struct tm *b)
 }
 
 /*
- *  call-seq:
- *    Time.utc(year) -> time
- *    Time.utc(year, month) -> time
- *    Time.utc(year, month, day) -> time
- *    Time.utc(year, month, day, hour) -> time
- *    Time.utc(year, month, day, hour, min) -> time
- *    Time.utc(year, month, day, hour, min, sec_with_frac) -> time
- *    Time.utc(year, month, day, hour, min, sec, usec_with_frac) -> time
- *    Time.utc(sec, min, hour, day, month, year, dummy, dummy, dummy, dummy) -> time
- *    Time.gm(year) -> time
- *    Time.gm(year, month) -> time
- *    Time.gm(year, month, day) -> time
- *    Time.gm(year, month, day, hour) -> time
- *    Time.gm(year, month, day, hour, min) -> time
- *    Time.gm(year, month, day, hour, min, sec_with_frac) -> time
- *    Time.gm(year, month, day, hour, min, sec, usec_with_frac) -> time
- *    Time.gm(sec, min, hour, day, month, year, dummy, dummy, dummy, dummy) -> time
- *
- *  Creates a Time object based on given values, interpreted as UTC (GMT). The
- *  year must be specified. Other values default to the minimum value
- *  for that field (and may be +nil+ or omitted). Months may
- *  be specified by numbers from 1 to 12, or by the three-letter English
- *  month names. Hours are specified on a 24-hour clock (0..23). Raises
- *  an ArgumentError if any values are out of range. Will
- *  also accept ten arguments in the order output by Time#to_a.
- *
- *  +sec_with_frac+ and +usec_with_frac+ can have a fractional part.
- *
- *     Time.utc(2000,"jan",1,20,15,1)  #=> 2000-01-01 20:15:01 UTC
- *     Time.gm(2000,"jan",1,20,15,1)   #=> 2000-01-01 20:15:01 UTC
+ * call-seq:
+ *   Time.utc(year, month=1, day=1, hour=0, min=0, sec_i=0, usec=0) -> new_time
+ *   Time.utc(sec_i, min, hour, day, month, year, dummy, dummy, dummy, dummy) -> new_time
+ *   Time.gm(year, month=1, day=1, hour=0, min=0, sec_i=0, usec=0) -> new_time
```

Comment:
I did that (to comply with the method_documentation.rdoc), but it caused call-seq for utc to be cited both for gm and utc. I didn't like that, so restored the status quo ante.  Solution?

Reply:
The correct solution would be to mark `gm` as a true alias (e.g. `rb_define_alias`), so it would just refer people to `utc`, assuming that RDoc handles that correctly.  However, I still think it is better for documentation for `gm` to only show `utc`, instead of showing both `utc` and `gm` for both.

Reply:
Should have said: I considered that (b/c it would do what I wanted), but was reluctant to make any non-doc change.

Reply:
@jeremyevans, I'm trying to alias as we discussed.
- This seems correct: ```rb_define_alias(rb_cTime, "mktime", "local")```.
- This is incorrect: ```rb_define_alias(rb_cTime, "gm", "utc")``` (hooked onto instance method, not singleton method).
- This seems better:  ```rb_define_alias(rb_singleton_class(rb_cTime), "gm", "utc")```, but may have implications I don't understand.

Advice, please?

Reply:
`rb_define_alias` defines aliases for the instance method of the class.  So if you are defining a class method alias, you need to operate on singleton class, as your last example shows.  `Time.local` and `Time.mktime` are class methods, so you would want to use the last form for that as well.  In your first example, you are defining `Time#mktime` as an alias of `Time#local`, which is not what you want.

There is existing code in Ruby just like your last example:

```c
rb_define_alias(rb_singleton_class(rb_mProcGID), "eid=", "grant_privilege");
```

Reply:
Aliased now in code.  Call-seq and examples removed.